libs/datamodel/parser-database/src/walkers/composite_type.rs

```diff
@@ -177,4 +188,17 @@ impl<'db> CompositeTypeFieldWalker<'db> {
             .and_then(|d| d.mapped_name)
             .map(|id| &self.db[id])
     }
+
+    /// Get the database name of the composite field.
+    pub fn get_field_database_name(self, field_id: ast::FieldId) -> &'db str {
+        self.db.types.composite_type_fields[&(self.ctid, field_id)]
+            .mapped_name
+            .map(|id| &self.db[id])
+            .unwrap_or_else(|| &self.db.ast[self.ctid][field_id].name.name)
+    }
+
+    /// The final database name of the field. See crate docs for explanations on database names.
+    pub(crate) fn database_name(self) -> &'db str {
+        self.default_mapped_name().unwrap_or_else(|| self.name())
```

Comment:
looks like it has to do with the `map` argument on the `@default()` attribute, not the `@map` on the field — we should check the tests